src/internal/visual-mode/__tests__/use-visual-mode.test.tsx

```diff
@@ -2,10 +2,16 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import React, { useRef } from 'react';
-import { useCurrentMode, useDensityMode, useReducedMotion } from '../index';
+import { useCurrentMode, useDensityMode, useReducedMotion, useVisualRefresh, clearVisualRefreshState } from '../index';
 import { render, screen } from '@testing-library/react';
 import { mutate } from './utils';
 
+const awsuiVisualRefreshFlag = Symbol.for('awsui-visual-refresh-flag');
+interface ExtendedWindow extends Window {
+  [awsuiVisualRefreshFlag]?: () => boolean;
+}
+declare const window: ExtendedWindow;
```

Comment:
I would recommend extracting `describe('useVisualRefresh', () => {...})` suite in a separate file to make these testing overrides more scoped